lib/atom-notes.js

```diff
@@ -183,19 +183,67 @@ function handleEvents (self) {
   const {openInterlink} = require('./interlink')
   const {CompositeDisposable, Disposable} = require('atom')
   self.subs = new CompositeDisposable()
-
+  
+  let previewCommand = null
+  
   // user commands
   self.subs.add(
     atom.commands.add('atom-workspace', 'atom-notes:toggle', () => {
       self.getNotesView().toggle()
     }),
-    atom.commands.add('atom-workspace', 'atom-notes:toggle-preview', () => {
-      self.getNotesView().toggle({preview: true})
-    }),
     atom.commands.add('atom-workspace', 'atom-notes:interlink', () => {
       openInterlink()
     })
   )
+  
+  /**
+   * Quickly activate the 'atom-notes:toggle-preview' command if not
+   * already active.
+   */
+  function activatePreviewCommand() {
+    if (!previewCommand) {
+      previewCommand = atom.commands.add(
+        'atom-workspace',
+        'atom-notes:toggle-preview',
+        () => self.getNotesView().toggle({preview: true}))
+      self.subs.add(previewCommand)
+    }
+  }
+  
+  /**
+   * Quickly deactivate the 'atom-notes:toggle-preview' command if is
+   * active.
+   */
+  function deactivatePreviewCommand() {
+    if (previewCommand) {
+      if (self.subs) {
+        self.subs.remove(previewCommand)
+      }
+      previewCommand.dispose()
+      previewCommand = null
+    }
+  }
+  
+  function activatePreview() {
+    if (atom.packages.isPackageActive('markdown-preview-plus') ||
+      atom.packages.isPackageActive('markdown-preview')) {
+      activatePreviewCommand()
+    } else {
+      deactivatePreviewCommand()
+    }
+  }
+  
+  /**
+   * Let's initialize on start
+   */
+  activatePreview()
+  
+  /**
+   * Check the packages again if something changed and activate or deactivate
+   * the command.
+   */
+  atom.packages.onDidActivatePackage(activatePreview)
```

Comment:
I think this will trigger a bunch of times when Atom first starts up. It would be best to delay adding this callback until after initial package activation is done.

Also `onDidActivatePackage()` returns a `Disposable` object that we need to save and be sure to `.dispose()` when we're done with it.